src/CLICommand/cli_command.ts

```diff
@@ -0,0 +1,86 @@
+import { Command } from 'commander';
+import { CliApiObject, ParsedArguments } from './cli';
+import { Parameter, ParameterName } from './parameter';
+
+export type CommandName = string;
+export interface CommandDescriptor {
+	name: CommandName;
+	parameters: ParameterName[];
+	action(options: ParsedArguments): void;
+}
+
+const program: CliApiObject = new Command() as CliApiObject;
+
+// Set up command line option parsing
+//const validActions = ['create-drive', 'rename-drive', 'upload-file'];
+program.option('-h, --help', 'Get help');
+//program.option('create-drive', 'action to create a new drive (and its corresponding root folder)');
+program.addHelpCommand(false);
+
+function setCommanderCommand(commandDescriptor: CommandDescriptor, program: CliApiObject): void {
```

Comment:
Maybe a small doc block and a small comment above the `const paramType` declaration here?  Its a little hard to follow the what and why here without digging around

Reply:
Adds JSDoc comment b0da89a